### ISSUE TITLE
Building on Power9

### ISSUE THREAD

**Issue by CloudyProton** [vegastrike/Vega-Strike-Engine-Source#666]
I have compiled master on a Power9 host but only after applying some tweaks.

Compilation normally quits with Error 2 
```
Vega-Strike-Engine-Source/engine/src/gfx/background.cpp:357:13: error: narrowing conversion of ‘-1’ from ‘int’ to ‘char’ [-Wnarrowing]
  357 |             };
      |   
```
But adding -Wno-narrowing to the compiler flags in ~/engine/CMakeLists.txt works around the build issue.

However, the affected code block in background.cpp is for rendering with multiple 2D texture faces. I think this may be affecting skybox rendering when running the game.
![vegastrike](https://user-images.githubusercontent.com/34248863/162347739-a4e8dadd-4b4b-4539-8e66-52fc8f807b79.png)
[CMakeError.log](https://github.com/vegastrike/Vega-Strike-Engine-Source/files/8447927/CMakeError.log)

**Comment by BenjamenMeyer** [comment]
Thanks for reporting. It's certainly worth a look. Power9 isn't on our official support list at this time; that said aside from CI availability (https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners) there is no issue having additional support for what the community wants to support. Let's first see if this is a more general bug, then we can look at adding options to the CMake file for Power9.

**Comment by stephengtuggy** [comment]
Wow, I'm always impressed when someone gets the game working (even partially) on a platform I've never heard of.

I'm pretty sure that this is one of many compiler warnings that we need to resolve. Not just on Power9, but on our other platforms as well. It's on my to-do list.

**Comment by eerykitty** [comment]
The problem is because the `char` type is signed by default on x86_64, but unsigned on ppc64le (and arm/aarch64). So [this line](https://github.com/vegastrike/Vega-Strike-Engine-Source/blob/master/engine/src/gfx/background.cpp#L238) needs to be changed to `signed char tcoord[4][4];`, and it works fine. I've been running this personally for a few years, but never got around to making a PR. Currently away from my P9 system so can't test it personally, but it shouldn't change behavior on x86(_64), but allows it to compile (and run correctly) on modern POWER and ARM systems.

**Comment by stephengtuggy** [comment]
Well, I tried testing with that change on Windows 10 x86_64. Unfortunately, I'm running into another, apparently unrelated issue. I'll have to fix that second issue before I can test this fix properly. I'll keep you updated.

**Comment by stephengtuggy** [comment]
OK, I was able to test @eerykitty 's fix on Windows 10 x86_64. It works fine, AFAICT. I'm including that fix in #670 .
